### ISSUE TITLE
Unable to link with multiple query params

### ISSUE THREAD

**Issue by JonZeolla** [executablebooks/MyST-Parser#760]
### Describe the bug

**context**
I am trying to make a link containing query params such as `https://example.com?first=a&second=b`

**expectation**
I expected that the rendered link would contain all of the query params.

**bug**
But instead the `&` becomes encoded as `&amp;` and breaks the link. I thought I'd be able to escape the `&` or that it would work natively, but neither seem to be true. 

### Reproduce the bug

```
[example][EXAMPLE]

[EXAMPLE]: https://example.com?first=a&second=b "Example"
```

and

```
[example][EXAMPLE]

[EXAMPLE]: https://example.com?first=a\&second=b "Example"
```

Renders:

```html
<a class="reference external" href="https://example.com?first=a&amp;second=b" title="Example">example</a>
```

### List your environment

Using MyST-Parser: 1.0.0

**Comment by sylvestre** [comment]
Reported also here: https://bugzilla.mozilla.org/show_bug.cgi?id=1828390

**Comment by chrisjsewell** [comment]
Heya, yep this is consistent with the CommonMark spec: 

- https://spec.commonmark.org/0.30/#example-502
- https://spec.commonmark.org/dingus/?text=%5Bexample%5D%5BEXAMPLE%5D%0A%0A%5BEXAMPLE%5D%3A%20https%3A%2F%2Fexample.com%3Ffirst%3Da%26second%3Db%20%22Example%22%0A

and also I believe with the HTML spec: https://stackoverflow.com/a/3705601/5033292

are you sure it is actually breaking the link, it doesn't appear to break for me in Firefox?

**Comment by JonZeolla** [comment]
It will load the page but the query param isn't properly sent to the server, so links like [this](https://www.youtube.com/watch?v=dQw4w9WgXcQ&t=2m12s) turn into [this](https://www.youtube.com/watch?v=dQw4w9WgXcQ\&amp\;t=2m12s) and don't process the start time.

**Comment by JonZeolla** [comment]
@chrisjsewell yeah, the backslash escapes were for GitHub rendering to make a link that matches when you click it (if you look at the resulting href on GitHub it is accurate)

**Comment by kan-fu** [comment]
I am having the exact same issue. For https://example.com?first=a&amp;second=b, the api service I am using complains that the second parameter is `amp;second`, which it expects to be `second`. Is there any way to let myst not encode the url?

**Comment by shimizukawa** [comment]
A potential solution would be to adjust the MyST-Parser code to prevent the initial escaping of the URL, allowing docutils or the final HTML writer to handle the necessary escaping. This could involve simply removing or modifying the call to escapeHtml in the MyST-Parser's conversion process.

```diff
diff --git a/myst_parser/mdit_to_docutils/base.py b/myst_parser/mdit_to_docutils/base.py
index 4a02459..cee67c9 100644
--- a/myst_parser/mdit_to_docutils/base.py
+++ b/myst_parser/mdit_to_docutils/base.py
@@ -996,7 +996,7 @@ class DocutilsRenderer(RendererProtocol):
             if "classes" in conversion:
                 ref_node["classes"].extend(conversion["classes"])
 
-        ref_node["refuri"] = escapeHtml(uri)
+        ref_node["refuri"] = uri
         if implicit_text is not None:
             with self.current_node_context(ref_node, append=True):
                 self.current_node.append(nodes.Text(implicit_text))
```


Additional Information:

The issue seems to stem from both the MyST-Parser and docutils/html writers, where the URL is being escaped more than once.
https://github.com/docutils/docutils/blob/b768e2626088711dec257b0847b563d02700a712/docutils/docutils/writers/_html_base.py

**Comment by amotl** [comment]
Hi there. We are observing the same.

1. We are able to confirm what @ailin-nemui was mentioning at https://github.com/executablebooks/MyST-Parser/issues/1028#issuecomment-2695539003:

   > This was still working in version myst-parser==0.18.1.

2. While this is not a generic workaround, but fragments of the conversation have been revolving around linking to YouTube, which is exactly also tripping us, check out this linking syntax, which works using just a single query parameter. ;]

   https://youtu.be/4xGUlG5Ck6c?t=105

**Comment by smipi1** [comment]
This also affects a lot of the documentation we are maintaining internally. How can we get #1126 good enough to be merged and released?

**Comment by krassowski** [comment]
> Heya, yep this is consistent with the CommonMark spec:
> ...
> and also I believe with the HTML spec: https://stackoverflow.com/a/3705601/5033292

Indeed, the HTML4 specification (https://www.w3.org/TR/html4/appendix/notes.html#h-B.2.2) required escaping ampersands:

> The URI that is constructed when a [form is submitted](https://www.w3.org/TR/html4/interact/forms.html#submit-format) may be used as an anchor-style link (e.g., the [href](https://www.w3.org/TR/html4/struct/links.html#adef-href) attribute for the [A](https://www.w3.org/TR/html4/struct/links.html#edef-A) element). Unfortunately, the use of the "&" character to separate form fields interacts with its use in SGML attribute values to delimit [character entity references](https://www.w3.org/TR/html4/intro/sgmltut.html#character-entities). For example, to use the URI "http://host/?x=1&y=2" as a linking URI, it must be written `<A href="http://host/?x=1&#38;y=2">` or `<A href="http://host/?x=1&amp;y=2">`.

However,
- it is not the case for HTML5 https://stackoverflow.com/questions/19441750/do-ampersands-still-need-to-be-encoded-in-urls-in-html5
- navigate to https://myst-parser.readthedocs.io/en/latest/?a=1&b=2&c=3 vs https://myst-parser.readthedocs.io/en/latest/?a=1&amp;b=2&amp;c=3 and execute `new URL(window.location.href).searchParams.get("b")` - only the former link will work; repeat with `new URLSearchParams(window.location.search).get("b")`, again only the former will work; I think this de facto the problem that users are facing
- GFM renders such links without encoding ampersands, this means this is only caught later on and not during review on GitHub
- the CommonMark spec indeed suggests encoding ampersands in URLs, but also says "Renderers may make different decisions about how to escape or normalize URLs in the output"

`URLSearchParams` behaviour follows the `application/x-www-form-urlencoded` parsing specification of the URL living standard from WHATWG (https://url.spec.whatwg.org/#urlencoded-parsing) and is what browsers implement:
> 1. Let sequences be the result of splitting input on 0x26 (&).

It looks like the tension is coming from the HTML 4 specification vs WHATWG URL specification. HTML 4 is designated as a Superseded Recommendation by W3C, thus it seems to me that it is perfectly acceptable and even preferable from the standard compliance point of view to NOT encode ampersands.

**Comment by krassowski** [comment]
Also, the new parser seems to handle it as users expect:

<img width="1042" height="349" alt="Image" src="https://github.com/user-attachments/assets/cc85fb7f-4d21-442e-8900-cff0be35316a" />

https://mystmd.org/sandbox?tab=demo&myst=SABlAHIAZQAgAGkAcwAgAGEAIABbAGwAaQBuAGsAXQAoAGgAdAB0AHAAcwA6AC8ALwBtAHkAcwB0AC0AcABhAHIAcwBlAHIALgByAGUAYQBkAHQAaABlAGQAbwBjAHMALgBpAG8ALwBlAG4ALwBsAGEAdABlAHMAdAAvAD8AYQA9AHgAJgBiAD0AeQAmAGMAPQB6ACkA

(as previously noted in https://github.com/jupyter-book/jupyter-book/issues/2271)
